### DISCUSSION TITLE
Where does this code keep the model files?

### DISCUSSION THREAD

**Discussion by gulzatmamytbek1** [PromtEngineer/localGPT#250] [Q&A]
I tested this code and it works, however, I did not see the actual model files ANYWHERE. Where are the model file?

**Comment by ziptron** [comment]
If you do not have a model downloaded, you can see where it gets the model from by running run_localGPT.py and finding this line:
"llama.cpp: loading model from /root/.cache/huggingface/hub/models--TheBloke--Llama-2-7B-Chat-GGML/snapshots/b616819cd4777514e3a2d9b8be69824aca8f5daf/llama-2-7b-chat.ggmlv3.q4_0.bin" It will show you where it stored it on your particular instance.

**Reply by EdSeymore** [reply]
Have a look in ~/.cache/huggingface/hub

**Comment by huycke** [comment]
I never figured this out. What I decided to do was download the models to a specific folder, then I pass the folder location with the model in the right file. Does anyone know if hugging face saves the models in a way that allows me to use them in different environments without having to download them once for each .conda or .venv I'm working in?
